src/lib/dip_switch.cpp

```diff
@@ -0,0 +1,42 @@
+/**
+ * @file dip_switch.cpp
+ *
+ * @brief Proxy DIP Switch class source
+ *
+ * @date 04/2024
+ */
+
+#ifndef MICRAS_PROXY_DIP_SWITCH_CPP
+#define MICRAS_PROXY_DIP_SWITCH_CPP
+
+#include "proxy/dip_switch.hpp"
+
+namespace proxy {
+template <uint8_t num_of_sensors>
+DipSwitch<num_of_sensors>::DipSwitch(const Config& config) {
+    for (uint8_t i = 0; i < num_of_switches; i++) {
+        this->subscribers[i] = config.node->create_subscription<std_msgs::msg::Bool>(
```

Comment:
qual a diferença entre usar `array.at(i)` e `array[i]`? pelo q lembro eh só que o `at` tem bound checking, mas posso estar maluco

fiquei curioso só pq em alguns lugares de outros construtores usaram o `at` e aqui usaram o `[]`...

apesar de ser mais lentinho eu acho o at mais maneiro kkkkk e esse construtor n parece precisar ser rápido ctor

Reply:
É isso ai mesmo, o `at` tem bound checking, ai a regra que eu costumo seguir é que se o índice q ta sendo acessado é fornecido pelo usuário, é bom usar `at`, mas se o índice for constante, ou se ta em um contexto controlado, tipo esse for ai, da pra usar só o `[]` mesmo. No contexto de embarcados tem um ponto extra né, pq o código é compilado com a [flag](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_exceptions.html) `-fno-exception`, ai sinceramente não sei o que acontece se o `at` falhar no bound checking.